app/code/Magento/Inventory/Test/Integration/Stock/GetProductQuantityInStockTest.php

```diff
@@ -0,0 +1,109 @@
+     */
+    private $reservationsAppend;
+
+    /**
+     * @var  ReservationCleanupInterface
+     */
+    private $reservationCleanup;
+
+    /**
+     * @var GetProductQuantityInStock
+     */
+    private $getProductQtyInStockService;
+
+    protected function setUp()
+    {
+        $this->indexer = Bootstrap::getObjectManager()->create(Indexer::class);
+        $this->indexer->load(StockItemIndexerInterface::INDEXER_ID);
+
+        $this->reservationBuilder = Bootstrap::getObjectManager()->get(ReservationBuilderInterface::class);
+        $this->reservationsAppend = Bootstrap::getObjectManager()->get(ReservationsAppendInterface::class);
+
+        $this->getProductQtyInStockService = Bootstrap::getObjectManager()->create(
+            GetProductQuantityInStockInterface::class
+        );
+
+        $this->reservationCleanup = Bootstrap::getObjectManager()->create(ReservationCleanupInterface::class);
+    }
+
+    public function tearDown()
+    {
+        /** @var IndexNameBuilder $indexNameBuilder */
+        $indexNameBuilder = Bootstrap::getObjectManager()->get(IndexNameBuilder::class);
+        /** @var IndexStructureInterface $indexStructure */
+        $indexStructure = Bootstrap::getObjectManager()->get(IndexStructureInterface::class);
+
+        foreach ([1, 2, 3] as $stockId) {
+            $indexName = $indexNameBuilder
+                ->setIndexId(StockItemIndexerInterface::INDEXER_ID)
+                ->addDimension('stock_', $stockId)
+                ->setAlias(Alias::ALIAS_MAIN)
+                ->create();
+            $indexStructure->delete($indexName);
+        }
+
+        // Cleanup reservations
+        $this->reservationCleanup->execute();
+    }
+
+    /**
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/products.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/sources.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/stocks.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/source_items.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/stock_source_link.php
+     */
+    public function testGetProductQuantity()
+    {
+        $this->indexer->reindexRow(1);
+
+        $reservations = [
+            $this->reservationBuilder->setStockId(1)->setSku('SKU-1')->setQuantity(-5)->build(), // reserve 5 units
+            $this->reservationBuilder->setStockId(1)->setSku('SKU-1')->setQuantity(1.5)->build(), // unreserve 1.5 units
+        ];
+        $this->reservationsAppend->execute($reservations);
+
+        $qty = $this->getProductQtyInStockService->execute('SKU-1', 1);
+        self::assertEquals(5, $qty);
+
+        $reservations = [
+            $this->reservationBuilder->setStockId(1)->setSku('SKU-1')->setQuantity(3.5)->build(), // unreserve 3.5 units
+        ];
+        $this->reservationsAppend->execute($reservations);
```

Comment:
why do we have this code
```
+        $reservations = [
 +            $this->reservationBuilder->setStockId(1)->setSku('SKU-1')->setQuantity(3.5)->build(), // unreserve 3.5 units
 +        ];
 +        $this->reservationsAppend->execute($reservations);
```
without any asserts afterwords ? 
is this kind of rollback action? which should prevent to affect further calculations?